src/wallet/wallet.cpp

```diff
@@ -3931,10 +3933,26 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
     watchonly_batch.reset(); // Flush
     // Do the removes
     if (txids_to_delete.size() > 0) {
-        if (ZapSelectTx(txids_to_delete) != DBErrors::LOAD_OK) {
```

Comment:
In ff214259de28216a5190c358b786306985f30227 "wallet: migration, remove watch-only transactions atomically"

It seems unnecessary to be making this change here as `ZapSelectTx` is now atomic.

Reply:
> In [ff21425](https://github.com/bitcoin/bitcoin/commit/ff214259de28216a5190c358b786306985f30227) "wallet: migration, remove watch-only transactions atomically"
> 
> It seems unnecessary to be making this change here as `ZapSelectTx` is now atomic.

True. This PR does not requires it. Will re-introduced it within #28574.